### PR TITLE
Add Poetry support

### PR DESCRIPTION
**Pull request by queue-miscreant**
Since it seems like the author wants to get this project into better shape, I might as well do my part by fixing up the package management.

This PR:
- Adds a `pyproject.toml` utilizing poetry (and generates a `poetry.lock` file)
- Moves source files into `dcolor/` and exposes names via `dcolor/__init__.py`
- Fixes examples, when run using `poetry run python`
  - Adding ".." to `sys.path` only works when running `python` from `examples/`. Poetry at least provides the guarantee of matching to the correct virtual environment, and works from anywhere in the project tree
- Runs [black](https://github.com/psf/black) to format all Python files in the project
- Removes the `__pycache__` and `.DS_Store` files (filesystem ephemera), and adds them to the `.gitignore`

### CODE REVIEW

File: dcolor/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from .dcolor import DColor
+from .hsvcolor import DColor as DColorHSV
+from .hsvcolor import DColor as DColorRGB
```
Comment (hernanat) [inline dcolor/__init__.py]: should this be:
```suggestion
from .rgbcolor import DColor as DColorRGB
```

File: README.md [hunk 2]
```diff
@@ -2,19 +2,39 @@
...
 
-## clone the repository
+```bash
+python3 -m pip install -U numpy
+python3 -m pip install -U matplotlib
+git clone https://github.com/hernanat/dcolor.git
+```
 
-`git clone https://github.com/hernanat/dcolor.git`
+### New (`poetry`)
```
Comment (hernanat) [inline README.md:21]: ty

**Comment by hernanat** [comment]
thanks! admittedly I don't know much about the python ecosystem these days, so was asking some colleagues for their input on this as well. just one comment

File: dcolor/__init__.py (hunk 1 above)
Comment (queue-miscreant) [reply dcolor/__init__.py]: Whoops. I fixed that in some other branches I've been working on.
...that said, I actually ended up retooling having them in separate classes entirely, then spent way too much time working on getting nice curves for an extra plotting variant. I'll have separate requests for them.

File: README.md (hunk 2 above)
Comment (queue-miscreant) [reply README.md:21]: No problem! I'm actually new to Poetry too, as of this year.

One thing you may have noticed further down is the markdown comment about generating `dist/`, which, along with virtualenvs, is one of the biggest things Poetry helps with. I'm admittedly rather weak when it comes to Python distribution, but it doesn't seem common to include wheelfiles in the Git project structure (at least, there aren't any in [matplotlib](https://github.com/matplotlib/matplotlib)).

**Comment by queue-miscreant** [comment]
Incidentally, while working on one of said other branches, I got interested in generating the images in the README programmatically. It seems like `images/hsvcolor.png` is incorrect. When I generate it, I get something identical to `dcolor.png` visually (although the binary contents differ).
image

Tool used: https://www.diffchecker.com/image-compare/

**Comment by queue-miscreant** [comment]
@hernanat Wanted to make sure you saw the commits I added to this.
I also left a big list of things I suggested working  toward in the issue linked above (#5). I have a branch based on this one which has a more imperative feel (and makes it easier to generate examples for the README), but I don't want to make this PR any harder to review.

**Comment by hernanat** [comment]
> @hernanat Wanted to make sure you saw the commits I added to this. I also left a big list of things I suggested working toward in the issue linked above (#5). I have a branch based on this one which has a more imperative feel (and makes it easier to generate examples for the README), but I don't want to make this PR any harder to review.

hey there! Thank you, yes I did see. Sorry -- I have been on vacation & traveling a bunch. I will try to get to all of this in the next few days when I have some free time.

**Comment by hernanat** [comment]
thank you again for all of this!

**Pull request merged**